src/java.desktop/share/classes/javax/swing/text/html/HTMLEditorKit.java

```diff
@@ -455,12 +455,10 @@ public StyleSheet getStyleSheet() {
         if (defaultStyles == null) {
             defaultStyles = new StyleSheet();
             appContext.put(DEFAULT_STYLES_KEY, defaultStyles);
-            try {
-                InputStream is = HTMLEditorKit.getResourceAsStream(DEFAULT_CSS);
+            try (InputStream is = HTMLEditorKit.getResourceAsStream(DEFAULT_CSS)) {
                 Reader r = new BufferedReader(
                         new InputStreamReader(is, ISO_8859_1));
                 defaultStyles.loadRules(r, null);
-                r.close();
```

Comment:
the reader was not added to the try block, it will not be closed.

Reply:
updated

Reply:
What about "new InputStreamReader()" here and below?

Reply:
added it to `try` too